bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -168,8 +168,11 @@ public void removed(Provider<E> provider, E element) {
         Collection<E> elements = elementMap.get(provider);
         if (elements != null) {
             try {
-                onRemoveElement(element);
-                elements.remove(element);
+                // the given "element" might not be the live instance but
+                // loaded from storage. operate on the real element:
+                E existingElement = get(element.getUID());
+                onRemoveElement(existingElement);
+                elements.remove(existingElement);
                 notifyListenersAboutRemovedElement(element);
```

Comment:
Just a quick first browse-through: Shouldn't `notifyListenersAboutRemovedElement(...)` also be called with `existingElement`?

Reply:
absolutely!

Reply:
I edited the lines once or twice to see the effect of the test. And while I'm typing this: The tests could also check that the listeners are called with the real entity. wdyt?

Reply:
done :-)